docs/4. repository/2. file-repositories.mdx

```diff
@@ -0,0 +1,135 @@
+
+### RepositoryV2
+
+RepositoryV2 is the format where we normalize the file names with their keccak256 hashes (source files must have a `keccak256` field in the metadata). So the example above would look like this:
+
+```
+contracts/full_match/1/0x801f3983c7baBF5E6ae192c84E1257844aDb4b4D/
+├── metadata.json
+└── sources/
+    └── 0xa38ec4e151e4d397d05bdfb94e6e4eb91e57a9fca3bc1c655289a4adf31a58fa.sol
+    └── 0xea27a3e2c4179a064caf9fe9a198addd526fd1d1ea467ea474a0c069e6eac957.sol
+```
+
+The files are exactly the same so their IPFS hashes will not change, and you can look up the metadata file to find the original path-alike source unit names.
+
+## IPFS
+
+Unfortunatelly publishing under IPNS is temporarily disabled. This is because of the difficulty of managing the whole filesystem over IPFS (with MFS etc.) and updating the IPNS regularly. 
+
+We still pin all the files on IPFS so you can access them over their individual CIDs (e.g. [`QmVij3h9z536ZG5cRpUmTfdoN9KR1Xp4ix2P7to9dPHgE5`](https://ipfs.io/ipfs/QmVij3h9z536ZG5cRpUmTfdoN9KR1Xp4ix2P7to9dPHgE5)).
+
+Look at the [Download section](#download) to learn how to download the whole repository.
+
+## Web
+
+Moved to [repo.sourcify.dev](/docs/repository/repo.sourcify.dev).
+
+## Download
+
+We compress the **RepositoryV2** weekly and publish on Cloudflare R2 under https://repo-backup.sourcify.dev ( `.dev` redirects to `.app` domain, which also belongs to Sourcify). 
+
+<TotalRepoSize/>
+
+[repo-backup.sourcify.dev](https://repo-backup.sourcify.dev) will redirect to a `manifest.json` file:
+
+<details>
+<summary>manifest.json</summary>
+
+```json
+{
+  "description": "Manifest file for when the Sourcify file repository was uploaded",
+  "timestamp": 1726030203254,
+  "dateStr": "2024-09-11T04:50:03.254904Z",
+  "files": [
+    {
+      "path": "sourcify-repository-2024-09-10T13-36-47/sourcify-repository-2024-09-10T13-36-47.part.gz.aa",
+      "sizeInBytes": 2097152000
+    },
+    ...
+    {
+      "path": "sourcify-repository-2024-09-10T13-36-47/sourcify-repository-2024-09-10T13-36-47.part.gz.ap",
+      "sizeInBytes": 800472503
+    }
+  ]
+}
+```
+</details>
+
+You can download all files in the `files` array and unzip them:
+
+1. Download the manifest file (`-L` to follow redirects):
+   ```bash
+   curl -L -O https://repo-backup.sourcify.dev/manifest.json
+   ```
+
+2. Extract file paths and download each file:
+   ```bash
+   jq -r '.files[].path' manifest.json | xargs -I {} curl -L -O https://repo-backup.sourcify.dev/{}
+   ```
+
+3. Concatenate the downloaded parts:
+   ```bash
+   cat sourcify-repository-*.part.gz.* > sourcify-repository.gz
+   ```
+
+4. Unzip the concatenated file:
+   ```bash
+   gunzip sourcify-repository.gz
+   ```
```

Comment:
I didn't have to do this step. I would directly call 

`tar -xvzf sourcify-repository.gz`

`z`: tells tar to decompress the archive using gzip